tests/FSharp.Compiler.ComponentTests/EmittedIL/Inlining/Match01_RealInternalSignatureOn.fs.il.net472.debug.bsl

```diff
@@ -955,40 +604,171 @@
+      IL_00a0:  stloc.s    V_9
+      IL_00a2:  ldarg.1
+      IL_00a3:  castclass  Match01/Test1/X13
+      IL_00a8:  stloc.s    V_10
+      IL_00aa:  call       class [runtime]System.Collections.IComparer [FSharp.Core]Microsoft.FSharp.Core.LanguagePrimitives::get_GenericComparer()
+      IL_00af:  stloc.s    V_4
+      IL_00b1:  ldloc.s    V_9
+      IL_00b3:  ldfld      int32 Match01/Test1/X13::item
+      IL_00b8:  stloc.s    V_5
+      IL_00ba:  ldloc.s    V_10
+      IL_00bc:  ldfld      int32 Match01/Test1/X13::item
+      IL_00c1:  stloc.s    V_6
+      IL_00c3:  ldloc.s    V_5
+      IL_00c5:  ldloc.s    V_6
+      IL_00c7:  cgt
+      IL_00c9:  ldloc.s    V_5
+      IL_00cb:  ldloc.s    V_6
+      IL_00cd:  clt
+      IL_00cf:  sub
+      IL_00d0:  ret
+
+      IL_00d1:  ldarg.0
+      IL_00d2:  castclass  Match01/Test1/X14
+      IL_00d7:  stloc.s    V_11
+      IL_00d9:  ldarg.1
+      IL_00da:  castclass  Match01/Test1/X14
+      IL_00df:  stloc.s    V_12
+      IL_00e1:  call       class [runtime]System.Collections.IComparer [FSharp.Core]Microsoft.FSharp.Core.LanguagePrimitives::get_GenericComparer()
+      IL_00e6:  stloc.s    V_4
+      IL_00e8:  ldloc.s    V_11
+      IL_00ea:  ldfld      int32 Match01/Test1/X14::item
+      IL_00ef:  stloc.s    V_5
+      IL_00f1:  ldloc.s    V_12
+      IL_00f3:  ldfld      int32 Match01/Test1/X14::item
+      IL_00f8:  stloc.s    V_6
+      IL_00fa:  ldloc.s    V_5
+      IL_00fc:  ldloc.s    V_6
+      IL_00fe:  cgt
+      IL_0100:  ldloc.s    V_5
+      IL_0102:  ldloc.s    V_6
+      IL_0104:  clt
+      IL_0106:  sub
+      IL_0107:  ret
+
+      IL_0108:  ldloc.0
+      IL_0109:  ldloc.1
+      IL_010a:  sub
+      IL_010b:  ret
+    } 
+
     .method public hidebysig virtual final instance int32  CompareTo(class Match01/Test1 obj) cil managed
     {
       .custom instance void [runtime]System.Runtime.CompilerServices.CompilerGeneratedAttribute::.ctor() = ( 01 00 00 00 ) 
       
-      .maxstack  4
-      .locals init (class [FSharp.Core]Microsoft.FSharp.Core.FSharpFunc`2<class [FSharp.Core]Microsoft.FSharp.Core.Unit,int32> V_0)
+      .maxstack  8
       IL_0000:  ldarg.0
-      IL_0001:  brfalse.s  IL_001a
+      IL_0001:  brfalse.s  IL_0011
 
       IL_0003:  ldarg.1
-      IL_0004:  brfalse.s  IL_0018
+      IL_0004:  brfalse.s  IL_000f
 
       IL_0006:  ldarg.0
       IL_0007:  ldarg.1
-      IL_0008:  newobj     instance void Match01/Test1/clo@4::.ctor(class Match01/Test1,
-                                                                    class Match01/Test1)
-      IL_000d:  stloc.0
-      IL_000e:  ldloc.0
-      IL_000f:  ldnull
-      IL_0010:  tail.
-      IL_0012:  callvirt   instance !1 class [FSharp.Core]Microsoft.FSharp.Core.FSharpFunc`2<class [FSharp.Core]Microsoft.FSharp.Core.Unit,int32>::Invoke(!0)
-      IL_0017:  ret
+      IL_0008:  ldnull
+      IL_0009:  call       int32 Match01/Test1::CompareTo$cont@4(class Match01/Test1,
```

Comment:
I assume this diff is THE thing we have wanted to address in the dotnet/performance regression with structs, right?
i.e. calling a function directly as opposed to going via new closure allocation+invocation.

Reply:
I think the merge of main fixed this. there is no longer a change in this baseline